src/binary-reader-ir.cc

```diff
@@ -824,6 +824,13 @@ Result BinaryReaderIR::OnFunctionName(Index index, StringSlice name) {
 
   Func* func = module->funcs[index];
   std::string dollar_name = std::string("$") + string_slice_to_string(name);
+  if (module->func_bindings.count(dollar_name) != 0) {
+    std::string orig_name = dollar_name;
+    int counter = 1;
+    do {
+      dollar_name = orig_name + "." + std::to_string(counter++);
+    } while (module->func_bindings.count(dollar_name) != 0);
+  }
   func->name = dup_string_slice(string_to_string_slice(dollar_name));
   module->func_bindings.emplace(dollar_name, Binding(index));
   return Result::Ok;
```

Comment:
We are allowing duplicate names since there is a real world screnario where this is possible (multiple compilations units have the same local function names).  I don't know of any place where allowing params with the same name would be useful/desirable.